### PR TITLE
feat(gatsby): Send config keys over IPC

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Send certain config keys over IPC to Gatsby Cloud.

- [x] Remove config file from gatsby-plugin-gatsby-cloud

## Related Issues

[ch44027]

### CODE REVIEW

**Comment by wardpeet** [comment]
For pathprefix we are saving a file to disk:
https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-gatsby-cloud/src/create-site-config.js

I'm okay with IPC but this already exists so it's easier on the cloud side

**Comment by LekoArts** [comment]
@dmccraw et al decided in the meeting that we'll do IPC instead of the file (and then later can deprecate that file)

**Comment by wardpeet** [comment]
Then we should remove it from the plugin using this PR - else it's never going to be removed 👍

**Review by dmccraw — APPROVED** [review]
:ship: it!

File: packages/gatsby/src/utils/gatsby-cloud-config.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+type ConstructConfigObjectResponse = Pick<
+  IGatsbyConfig,
+  "trailingSlash" | "assetPrefix" | "pathPrefix"
+>
+
+export function conststructConfigObject(
+  gatsbyConfig: IGatsbyConfig
+): ConstructConfigObjectResponse {
+  return {
+    trailingSlash: gatsbyConfig.trailingSlash ?? `legacy`,
```
Comment (EzraEllette) [inline packages/gatsby/src/utils/gatsby-cloud-config.ts:12]: Does `trailingSlash` have a `legacy` value? How does it compare with `ignore`?

Comment (LekoArts) [reply packages/gatsby/src/utils/gatsby-cloud-config.ts:12]: Yes, it has a `legacy` value. On Cloud it should behave the same as `ignore` for now, we'll remove `legacy` in Gatsby v5

**Pull request merged**